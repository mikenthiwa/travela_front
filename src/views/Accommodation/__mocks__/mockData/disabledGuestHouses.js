const disabledGuestHouses = [
  {
    id: 'juhiuhoji0',
    houseName: 'Guest House A',
    location: 'Nairobi, Kenya',
    bathRooms: 5,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'bvuh9iei8bh',
        roomName: 'ZAPTIU',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'NBiojnn98u',
        roomName: 'AFEIK',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'jbg8oijjolp',
    houseName: 'Guest House B',
    location: 'Nairobi, Kenya',
    bathRooms: 3,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'bnvjhijkpj',
        roomName: 'APLI',
        roomType: 'Single',
        bedCount: 1,
        faulty: false,
      },
      {
        id: 'bnvuiohj',
        roomName: 'Onij',
        roomType: 'Single',
        bedCount: 4,
        faulty: false,
      }
    ]
  },
  {
    id: 'nbuiojkpl',
    houseName: 'Guest House C',
    bathRooms: 4,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Lagos, Nigeria',
    rooms: [
      {
        id: 'bvuijkok',
        roomName: 'Two',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
    ]
  },
  {
    id: 'nmbkihjokko',
    houseName: 'Guest House D',
    bathRooms: 1,
    disabled: true,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Kigali, Rwanda',
    rooms: [
      {
        id: 'jbihjjkpk',
        roomName: 'Roof',
        roomType: 'Suite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
];

export default disabledGuestHouses;
