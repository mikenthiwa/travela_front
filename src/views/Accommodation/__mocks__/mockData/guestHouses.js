const guestHouses = [
  {
    id: 'wrtiuy56_u',
    houseName: 'Guest House 1',
    location: 'Nairobi, Kenya',
    bathRooms: 5,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'tryuompl0',
        roomName: 'Nyati',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'trytrimo',
        roomName: 'Maputi',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'wrtnytom_9',
    houseName: 'Guest House 2',
    location: 'Nairobi, Kenya',
    bathRooms: 3,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    rooms: [
      {
        id: 'plmyt0pol',
        roomName: 'Santana',
        roomType: 'Single',
        bedCount: 1,
        faulty: false,
      },
      {
        id: 'plmmnuyt',
        roomName: 'Shiloh',
        roomType: 'Single',
        bedCount: 4,
        faulty: false,
      }
    ]
  },
  {
    id: 'bytrwqjk9',
    houseName: 'Guest House 3',
    bathRooms: 4,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Lagos, Nigeria',
    rooms: [
      {
        id: 'btrfdseq23',
        roomName: 'Aben',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      },
      {
        id: 'btrfdsnytr3',
        roomName: 'Aben',
        roomType: 'Ensuite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
  {
    id: 'requntm0',
    houseName: 'Guest House 4',
    bathRooms: 1,
    imageUrl: 'http://images/guest-house.jpg',
    genderPolicy: 'unisex',
    location: 'Kigali, Rwanda',
    rooms: [
      {
        id: 'mnytbvfdc9',
        roomName: 'Berger',
        roomType: 'Suite',
        bedCount: 2,
        faulty: false,
      }
    ]
  },
];

export default guestHouses;
