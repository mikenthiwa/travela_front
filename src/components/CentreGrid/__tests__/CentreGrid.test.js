import React from 'react';
import { shallow } from 'enzyme';
import WithLoadingCentreGrid, { CentreGrid } from '..';
import guestHouses from '../../../views/Accommodation/__mocks__/mockData/guestHouses';
import disabledGuestHouses from '../../../views/Accommodation/__mocks__/mockData/disabledGuestHouses';

const props = {
  guestHouses,
  disabledGuestHouses,
  handleOnRestore: () => null
};

describe('<CentreGrid />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CentreGrid {...props} />);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the right number of centres', () => {
    expect(wrapper.find('CentreCard').length).toBe(8);
  });

  it('renders the placeholder when the app is loading', () => {
    wrapper = shallow(<WithLoadingCentreGrid {...{...props, isLoading: true}} />);
    expect(wrapper.find('ResidencePlaceholder').length).toBe(1);
    expect(wrapper.find('CentreCard').length).toBe(0);
  });

  it('renders a div with text when there are no centres', () => {
    wrapper.setProps({
      guestHouses: []
    });
    expect(wrapper.find('CentreCard').length).toBe(4);
    expect(wrapper.find('.table__requests--empty').text())
      .toEqual('No accommodation centres at the moment');
  });

  it('renders a div with text when there are no centres', () => {
    wrapper.setProps({
      guestHouses: []
    });
    expect(wrapper.find('CentreCard').length).toBe(4);
    expect(wrapper.find('.table__requests--empty').text())
      .toEqual('No accommodation centres at the moment');
  });

  it(`renders a div with error when there is an
    error fetching accommodation`, () => {
    wrapper.setProps({
      guestHouses: null,
      error: 'Error fetching accommodation'
    });
    expect(wrapper.find('CentreCard').length).toBe(4);
    expect(wrapper.find('.table__requests--error').text())
      .toEqual('Error fetching accommodation');
  });

  it('passes the right props to the CentreCard component', () => {
    const expectedProps = {
      bathrooms: 3,
      beds: 5,
      cardImage:'http://images/guest-house.jpg',
      countryFlagImage: 'https://www.countryflags.io/KE/flat/64.png',
      guestHouse: {
        bathRooms: 3,
        houseName: 'Guest House 2',
        id: 'wrtnytom_9',
        imageUrl: 'http://images/guest-house.jpg',
        genderPolicy: 'unisex',
        location: 'Nairobi, Kenya',
        rooms: [{
          bedCount: 1,
          faulty: false,
          id: 'plmyt0pol',
          roomName: 'Santana',
          roomType: 'Single',
        }, {
          bedCount: 4,
          faulty: false,
          id: 'plmmnuyt',
          roomName: 'Shiloh',
          roomType: 'Single',
        }],
      },
      guestHouseLocation: 'Nairobi, Kenya',
      guestHouseName: 'Guest House 2',
      imageAlt: 'Guest House 2 image',
      guesthouseId: 'wrtnytom_9',
    };
    const secondExpectedProps = {
      bathrooms: 4,
      beds: 4,
      cardImage:'http://images/guest-house.jpg',
      countryFlagImage: 'https://www.countryflags.io/NG/flat/64.png',
      guestHouse: {
        bathRooms: 4,
        houseName: 'Guest House 3',
        id: 'bytrwqjk9',
        imageUrl: 'http://images/guest-house.jpg',
        genderPolicy: 'unisex',
        location: 'Lagos, Nigeria',
        rooms: [{
          bedCount: 2,
          faulty: false,
          id: 'btrfdseq23',
          roomName: 'Aben',
          roomType: 'Ensuite',
        }, {
          bedCount: 2,
          faulty: false,
          id: 'btrfdsnytr3',
          roomName: 'Aben',
          roomType: 'Ensuite',
        }],
      },
      guestHouseLocation: 'Lagos, Nigeria',
      guestHouseName: 'Guest House 3',
      imageAlt: 'Guest House 3 image',
      guesthouseId: 'bytrwqjk9',
      disabledGuestHouse: {}
    };
    const secondCentre = wrapper.find('CentreCard').at(1);
    const thirdCentre = wrapper.find('CentreCard').at(2);

    /* Using `toEqual` instead of `toMatchObject` fails the test as with the error that
    Compared values have no visual difference.
    */
    expect(secondCentre.props()).toMatchObject(expectedProps);
    expect(thirdCentre.props()).toMatchObject(secondExpectedProps);
  });
});
