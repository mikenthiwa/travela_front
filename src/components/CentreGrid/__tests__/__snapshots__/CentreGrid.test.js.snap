// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CentreGrid /> renders correctly 1`] = `
<Fragment>
  <div
    className="mdl-grid accommodation-grid"
  >
    <CentreCard
      bathrooms={5}
      beds={4}
      cardImage="http://images/guest-house.jpg"
      countryFlagImage="https://www.countryflags.io/KE/flat/64.png"
      disabledGuestHouse={Object {}}
      guestHouse={
        Object {
          "bathRooms": 5,
          "genderPolicy": "unisex",
          "houseName": "Guest House 1",
          "id": "wrtiuy56_u",
          "imageUrl": "http://images/guest-house.jpg",
          "location": "Nairobi, Kenya",
          "rooms": Array [
            Object {
              "bedCount": 2,
              "faulty": false,
              "id": "tryuompl0",
              "roomName": "Nyati",
              "roomType": "Ensuite",
            },
            Object {
              "bedCount": 2,
              "faulty": false,
              "id": "trytrimo",
              "roomName": "Maputi",
              "roomType": "Ensuite",
            },
          ],
        }
      }
      guestHouseLocation="Nairobi, Kenya"
      guestHouseName="Guest House 1"
      guesthouseId="wrtiuy56_u"
      handleOnRestore={[Function]}
      imageAlt="Guest House 1 image"
      key="wrtiuy56_u"
    />
    <CentreCard
      bathrooms={3}
      beds={5}
      cardImage="http://images/guest-house.jpg"
      countryFlagImage="https://www.countryflags.io/KE/flat/64.png"
      disabledGuestHouse={Object {}}
      guestHouse={
        Object {
          "bathRooms": 3,
          "genderPolicy": "unisex",
          "houseName": "Guest House 2",
          "id": "wrtnytom_9",
          "imageUrl": "http://images/guest-house.jpg",
          "location": "Nairobi, Kenya",
          "rooms": Array [
            Object {
              "bedCount": 1,
              "faulty": false,
              "id": "plmyt0pol",
              "roomName": "Santana",
              "roomType": "Single",
            },
            Object {
              "bedCount": 4,
              "faulty": false,
              "id": "plmmnuyt",
              "roomName": "Shiloh",
              "roomType": "Single",
            },
          ],
        }
      }
      guestHouseLocation="Nairobi, Kenya"
      guestHouseName="Guest House 2"
      guesthouseId="wrtnytom_9"
      handleOnRestore={[Function]}
      imageAlt="Guest House 2 image"
      key="wrtnytom_9"
    />
    <CentreCard
      bathrooms={4}
      beds={4}
      cardImage="http://images/guest-house.jpg"
      countryFlagImage="https://www.countryflags.io/NG/flat/64.png"
      disabledGuestHouse={Object {}}
      guestHouse={
        Object {
          "bathRooms": 4,
          "genderPolicy": "unisex",
          "houseName": "Guest House 3",
          "id": "bytrwqjk9",
          "imageUrl": "http://images/guest-house.jpg",
          "location": "Lagos, Nigeria",
          "rooms": Array [
            Object {
              "bedCount": 2,
              "faulty": false,
              "id": "btrfdseq23",
              "roomName": "Aben",
              "roomType": "Ensuite",
            },
            Object {
              "bedCount": 2,
              "faulty": false,
              "id": "btrfdsnytr3",
              "roomName": "Aben",
              "roomType": "Ensuite",
            },
          ],
        }
      }
      guestHouseLocation="Lagos, Nigeria"
      guestHouseName="Guest House 3"
      guesthouseId="bytrwqjk9"
      handleOnRestore={[Function]}
      imageAlt="Guest House 3 image"
      key="bytrwqjk9"
    />
    <CentreCard
      bathrooms={1}
      beds={2}
      cardImage="http://images/guest-house.jpg"
      countryFlagImage="https://www.countryflags.io/RW/flat/64.png"
      disabledGuestHouse={Object {}}
      guestHouse={
        Object {
          "bathRooms": 1,
          "genderPolicy": "unisex",
          "houseName": "Guest House 4",
          "id": "requntm0",
          "imageUrl": "http://images/guest-house.jpg",
          "location": "Kigali, Rwanda",
          "rooms": Array [
            Object {
              "bedCount": 2,
              "faulty": false,
              "id": "mnytbvfdc9",
              "roomName": "Berger",
              "roomType": "Suite",
            },
          ],
        }
      }
      guestHouseLocation="Kigali, Rwanda"
      guestHouseName="Guest House 4"
      guesthouseId="requntm0"
      handleOnRestore={[Function]}
      imageAlt="Guest House 4 image"
      key="requntm0"
    />
  </div>
  <div>
    <hr />
    <div
      className="disabled-title"
    >
      DISABLED GUESTHOUSES
    </div>
    <div
      className="mdl-grid disabled-accommodation"
    >
      <CentreCard
        bathrooms={5}
        beds={4}
        cardImage="http://images/guest-house.jpg"
        countryFlagImage="https://www.countryflags.io/KE/flat/64.png"
        disabledGuestHouse={
          Object {
            "bathRooms": 5,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House A",
            "id": "juhiuhoji0",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Nairobi, Kenya",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "bvuh9iei8bh",
                "roomName": "ZAPTIU",
                "roomType": "Ensuite",
              },
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "NBiojnn98u",
                "roomName": "AFEIK",
                "roomType": "Ensuite",
              },
            ],
          }
        }
        guestHouse={
          Object {
            "bathRooms": 5,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House A",
            "id": "juhiuhoji0",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Nairobi, Kenya",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "bvuh9iei8bh",
                "roomName": "ZAPTIU",
                "roomType": "Ensuite",
              },
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "NBiojnn98u",
                "roomName": "AFEIK",
                "roomType": "Ensuite",
              },
            ],
          }
        }
        guestHouseLocation="Nairobi, Kenya"
        guestHouseName="Guest House A"
        guesthouseId="juhiuhoji0"
        handleOnRestore={[Function]}
        imageAlt="Guest House A image"
        key="juhiuhoji0"
      />
      <CentreCard
        bathrooms={3}
        beds={5}
        cardImage="http://images/guest-house.jpg"
        countryFlagImage="https://www.countryflags.io/KE/flat/64.png"
        disabledGuestHouse={
          Object {
            "bathRooms": 3,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House B",
            "id": "jbg8oijjolp",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Nairobi, Kenya",
            "rooms": Array [
              Object {
                "bedCount": 1,
                "faulty": false,
                "id": "bnvjhijkpj",
                "roomName": "APLI",
                "roomType": "Single",
              },
              Object {
                "bedCount": 4,
                "faulty": false,
                "id": "bnvuiohj",
                "roomName": "Onij",
                "roomType": "Single",
              },
            ],
          }
        }
        guestHouse={
          Object {
            "bathRooms": 3,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House B",
            "id": "jbg8oijjolp",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Nairobi, Kenya",
            "rooms": Array [
              Object {
                "bedCount": 1,
                "faulty": false,
                "id": "bnvjhijkpj",
                "roomName": "APLI",
                "roomType": "Single",
              },
              Object {
                "bedCount": 4,
                "faulty": false,
                "id": "bnvuiohj",
                "roomName": "Onij",
                "roomType": "Single",
              },
            ],
          }
        }
        guestHouseLocation="Nairobi, Kenya"
        guestHouseName="Guest House B"
        guesthouseId="jbg8oijjolp"
        handleOnRestore={[Function]}
        imageAlt="Guest House B image"
        key="jbg8oijjolp"
      />
      <CentreCard
        bathrooms={4}
        beds={2}
        cardImage="http://images/guest-house.jpg"
        countryFlagImage="https://www.countryflags.io/NG/flat/64.png"
        disabledGuestHouse={
          Object {
            "bathRooms": 4,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House C",
            "id": "nbuiojkpl",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Lagos, Nigeria",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "bvuijkok",
                "roomName": "Two",
                "roomType": "Ensuite",
              },
            ],
          }
        }
        guestHouse={
          Object {
            "bathRooms": 4,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House C",
            "id": "nbuiojkpl",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Lagos, Nigeria",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "bvuijkok",
                "roomName": "Two",
                "roomType": "Ensuite",
              },
            ],
          }
        }
        guestHouseLocation="Lagos, Nigeria"
        guestHouseName="Guest House C"
        guesthouseId="nbuiojkpl"
        handleOnRestore={[Function]}
        imageAlt="Guest House C image"
        key="nbuiojkpl"
      />
      <CentreCard
        bathrooms={1}
        beds={2}
        cardImage="http://images/guest-house.jpg"
        countryFlagImage="https://www.countryflags.io/RW/flat/64.png"
        disabledGuestHouse={
          Object {
            "bathRooms": 1,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House D",
            "id": "nmbkihjokko",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Kigali, Rwanda",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "jbihjjkpk",
                "roomName": "Roof",
                "roomType": "Suite",
              },
            ],
          }
        }
        guestHouse={
          Object {
            "bathRooms": 1,
            "disabled": true,
            "genderPolicy": "unisex",
            "houseName": "Guest House D",
            "id": "nmbkihjokko",
            "imageUrl": "http://images/guest-house.jpg",
            "location": "Kigali, Rwanda",
            "rooms": Array [
              Object {
                "bedCount": 2,
                "faulty": false,
                "id": "jbihjjkpk",
                "roomName": "Roof",
                "roomType": "Suite",
              },
            ],
          }
        }
        guestHouseLocation="Kigali, Rwanda"
        guestHouseName="Guest House D"
        guesthouseId="nmbkihjokko"
        handleOnRestore={[Function]}
        imageAlt="Guest House D image"
        key="nmbkihjokko"
      />
    </div>
  </div>
</Fragment>
`;
